feat: added design doc template and short instructions (#551)
hashgraph/hedera-block-node: server/docs/design/how-to-design-doc.md

## How to start a design doc

> 1) Copy the template from `docs/design/design-doc-template.md` to a new markdown file.
> 2) Take a look at `docs/design/block-verification.md` as an example of the detail level and tone.

Reviewer: Linking to the files can be helpful...
```suggestion
1) Copy the [template](design-doc-template.md) from `docs/design/design-doc-template.md` to a new markdown file.
2) Take a look at [this example](block-verification.md) `docs/design/block-verification.md` for the detail level and tone.
```
